### DISCUSSION TITLE
Exposing Pytest fixtures

### DISCUSSION THREAD

**Discussion by noirbizarre** [pdm-project/pdm#1571] [Ideas]
Hi 👋🏼 

I have been working on some plugins and I often found myself duplicating the `pdm`'s `pytest` fixtures that I need to test my plugin.
I already started a branch earlier this year but I didn't submitted it and I wanted to know if it would be accetable and accepted to move most of the fixtures from `tests/conftest.py` into a reusable `pdm/pytest.py` fixture module ?
This results in `conftest.py` only needing this to benefit from all fixtures (especially the `invoke` fixture)
```python
pytest_plugins = [
    "pdm.pytest",
]
```

Because I have already done it once, I already know the consequences:
- some fixtures would be splitted between the generic behavior and the one expected by `pdm` test suite
- a `pytest` extra (aka. a new entry in `project.optionnal-dependencies`) would be added for those wanting access to the fixture in their own project
- a few fixtures renaming to be more consistent for usage out of `pdm` codebase context
- exposed as a module and not a pytest plugin (both are officialy supported way of exposing resulable pytest fixtures) because loading order change when you expose a plugin and this has some sideeffects on PDM test suite.

A legacy version of this is available here: https://github.com/noirbizarre/pdm/tree/feature/pytest (not up to date, still some WIP but gives an overhaul idea of the refactoring)

Should I go further (update it, clean it...) and submit it as PR or do you think this is overengineered and you prefer each plugin to copy-paste the needed fixtures tree ?

**Comment by frostming** [comment]
> Should I go further (update it, clean it...) and submit it as PR or do you think this is overengineered and you prefer each plugin to copy-paste the needed fixtures tree ?

LGTM, it would be nicer to be a PR.
